cilium-cli/connectivity/tests/pod.go

```diff
@@ -253,3 +259,176 @@ func (s *podToPodNoFrag) Run(ctx context.Context, t *check.Test) {
 
 	})
 }
+
+func PodToPodMissingIPCache(opts ...Option) check.Scenario {
+	options := &labelsOption{}
+	for _, opt := range opts {
+		opt(options)
+	}
+	return &podToPodMissingIPCache{
+		sourceLabels:      options.sourceLabels,
+		destinationLabels: options.destinationLabels,
+		method:            options.method,
+	}
+}
+
+type podToPodMissingIPCache struct {
+	sourceLabels      map[string]string
+	destinationLabels map[string]string
+	method            string
+}
+
+func (s *podToPodMissingIPCache) Name() string {
+	return "pod-to-pod-missing-ipcache"
+}
+
+type bpfMap struct {
+	ID int `json:"id"`
+}
+
+type bpfMapLookup struct {
+	Key   []string `json:"key"`
+	Value []string `json:"value"`
+}
+
+func dropCountByUnencrypted(ctx context.Context, ct *check.ConnectivityTest) (count int, err error) {
+	unencryptedCountCmd := []string{
+		"/bin/sh", "-c",
+		`cilium metrics list -o json | jq '.[] | select((.name == "cilium_drop_count_total") and (.labels.reason | IN("Traffic is unencrypted"))) | .value'`,
+	}
+
+	for _, ciliumPod := range ct.CiliumPods() {
+		output, err := ciliumPod.K8sClient.ExecInPod(ctx, ciliumPod.Pod.Namespace, ciliumPod.Pod.Name, defaults.AgentContainerName, unencryptedCountCmd)
+		if err != nil {
+			return 0, err
+		}
+		if output.String() != "" {
+			c, err := strconv.Atoi(strings.TrimSpace(output.String()))
+			if err != nil {
+				return 0, err
+			}
+			count += c
+		}
+	}
+	return
+}
+
+func structToStrSlice(key *ipcachemap.Key) []string {
+	ptr := unsafe.Pointer(reflect.ValueOf(key).Pointer())
+	size := unsafe.Sizeof(*key)
+	byteSlice := make([]string, size)
+	for i := 0; i < int(size); i++ {
+		byteSlice[i] = fmt.Sprintf("0x%02x", *(*byte)(unsafe.Pointer(uintptr(ptr) + uintptr(i))))
+	}
+	return byteSlice
```

Comment:
Any reason we are not using `encoding/binary.Write` here? https://pkg.go.dev/encoding/binary#Write